### PR TITLE
feature/AK-67: Added AWS 'VPC Endpoint Cross Account Access' plugin and t…

### PR DESCRIPTION
**Pull request by AkhtarAmir**
…est cases

### CODE REVIEW

File: helpers/aws/functions.js [hunk 1]
```diff
@@ -217,14 +218,19 @@ function crossAccountPrincipal(principal, accountId) {
         awsPrincipals = [awsPrincipals];
     }
 
+    var principals = [];
+    var response = false;
+
     for (var a in awsPrincipals) {
-        if (/^arn:aws:iam::[0-9]{12}.*/.test(awsPrincipals[a]) &&
+        if (/^arn:aws:iam|sts::[0-9]{12}.*/.test(awsPrincipals[a]) &&
```
Comment (giorod3) [inline helpers/aws/functions.js:225]: double check this regex. I tried it and it wasn't working this way. I added parenthesis and it worked

```suggestion
        if (/^arn:aws:(iam|sts)::[0-9]{12}.*/.test(awsPrincipals[a]) &&
```

Comment (AkhtarAmir) [reply helpers/aws/functions.js]: Yes, you are right. It was working that way as well but this way is better

File: helpers/aws/functions.js [hunk 2]
```diff
@@ -217,14 +218,19 @@ function crossAccountPrincipal(principal, accountId) {
...
+        if (/^arn:aws:(iam|sts)::[0-9]{12}.*/.test(awsPrincipals[a]) &&
             awsPrincipals[a].indexOf(accountId) === -1) {
-            return true;
+            if (!fetchPrincipals) return true;
+            principals.push(awsPrincipals[a]);
         }
     }
 
-    return false;
+    if (fetchPrincipals) response = principals;
```
Comment (giorod3) [inline helpers/aws/functions.js:232]:
```suggestion
    if (fetchPrincipals) return principals;
```

File: helpers/aws/functions.js [hunk 3]
```diff
@@ -217,14 +218,19 @@ function crossAccountPrincipal(principal, accountId) {
...
+    return response;
```
Comment (giorod3) [inline helpers/aws/functions.js:233]:
```suggestion
    return false;
```

**Review by giorod3 — CHANGES_REQUESTED** [review]
just the minor changes

**Review by giorod3 — APPROVED** [review]

**Pull request merged**